### PR TITLE
[Docs] Taking fixpoints of polymorphic mutually recursive functions

### PR DESCRIPTION
**Pull request by effectfully**

### CODE REVIEW

File: docs/fomega/mutual-term-level-recursion/FixN.agda [hunk 1]
```diff
@@ -356,6 +356,38 @@ module LazinessStrictness where
...
+
+    poly-mono-by2' : (∀ {Q} -> F Q -> Q) -> {R : Set} -> F R -> R
+    poly-mono-by2' r k = k (λ x₁ -> r λ f₁ f₂ -> f₁ x₁) (λ x₂ -> r λ f₁ f₂ -> f₂ x₂)
+
+    poly-mono-fix2' : (∀ {Q} -> F Q -> F Q) -> {R : Set} -> F R -> R
+    poly-mono-fix2' = fixBy poly-mono-by2'
+
+  -- So in Plutus Core we can just generate appropriate `by*` functions on the fly depending on
+  -- the type schema of each of the functions that we take the fixed point of, while compiling
+  -- those functions from Plutus IR.
```
Comment (michaelpj) [inline docs/fomega/mutual-term-level-recursion/FixN.agda]: Of course, this isn't what we do at the moment. We do what's described in the paper, namely "thunking" the recursive values by making them functions from `()` and applying them at all the uses inside the body.

This is nice and avoids the `()`s, though.

Comment (effectfully) [reply docs/fomega/mutual-term-level-recursion/FixN.agda]: Yes, I didn't realize we could generate `by`s on the fly for polymorphic functions at the time.

Comment (effectfully) [reply docs/fomega/mutual-term-level-recursion/FixN.agda]: Which is rather unfortunate. I think, it'd make the paper a tiny bit better.

Comment (michaelpj) [reply docs/fomega/mutual-term-level-recursion/FixN.agda]: Yes, although the thunking transformation has the advantage that it works for anything of non-function type, not just polymorphic functions.

Comment (effectfully) [reply docs/fomega/mutual-term-level-recursion/FixN.agda]: As far as I remember, you haven't shown anything else besides polymorphic functions that can work in our strict setting. And we literally compile everything to functions (sometimes sprinkled with `iwrap` and `unwrap`). So as before I find this argument rather weak given that we do not have a single example to illustrate your point. Or do we?

Comment (michaelpj) [reply docs/fomega/mutual-term-level-recursion/FixN.agda]: Sure, but sometimes you add things and robustness against future changes is also useful.

**Pull request merged**